api-get-object.go

```diff
@@ -132,8 +217,32 @@ type Object struct {
 	// Keeps track of closed call.
 	isClosed bool
 
+	// Keeps track of if this is the first call.
+	isStarted bool
+
 	// Previous error saved for future calls.
 	prevErr error
+
+	// Keeps track of if this object has been read yet.
+	beenRead bool
+}
+
+// setObjectInfo - blocks until the first request is completed and then either errors out
+// or sets the received objectInfo.
+func (o *Object) setObjectInfo() (int, error) {
```

Comment:
setObjectInfo should only set the objectInfo and return error if any. It shouldn't return size. Size is available for future operations via ``o.objectInfo``